source/extensions/filters/http/grpc_http1_reverse_bridge_transcoder/filter.cc

```diff
@@ -243,10 +240,12 @@ Http::FilterHeadersStatus Filter::encodeHeaders(Http::ResponseHeaderMap& headers
   headers.setContentType(Http::Headers::get().ContentTypeValues.Grpc);
   headers.setReferenceContentType(Http::Headers::get().ContentTypeValues.Grpc);
 
-  // TODO: Adjust Content Length at a later point (maybe use Trailers for this?)
-  // adjustContentLength(headers, [](auto length) { return length + Grpc::GRPC_FRAME_HEADER_SIZE;
-  // }); headers.setContentLength(16);
-  headers.removeContentLength();
+  // NOTE: Content length handling. Since we don't know the content length
+  // before body transcoding. Memorize a pointer to the header map and use it
+  // in decodeData().
+  encoder_headers_ = &headers;
+
+  // TODO: Do we need this GRPC Status thing?
```

Comment:
I think it is need when error is returned, with(out) buffer. Have you tried that? Do we have test for this?

Reply:
Not at the moment. I guess it is one extra Task for the Test implementation.
Anyway, since the status is calculated from the headers and we have a pointer accessed in the encodeData method, we can determine the status right there. There is no need memoize the value.

Reply:
Resolved in latest PR